### PR TITLE
Handle absolute leafrefs being used as keys for a list.

### PR DESCRIPTION
**Pull request by robshakir**
This PR addresses #489, where a panic was thrown where
an absolute path was used in a leafref that was the key
of a list.

This change adds:

```
* (A) integration_tests/schemaops
  - Add a new integration test that works against an
    arbitrary schema to allow complex test cases to
    be reproduced and part of ongoing integration
    tests.
* (M) generator/.gitignore
  - Add a gitignore to ignore built binaries.
* (M) .gitignore
  - Avoid .vscode folders being created.
* (M) ytypes/list(_test).go
  - Ensure that we handle the absolute leafrefs
    correctly.
```

### CODE REVIEW

File: ytypes/list.go [hunk 1]
```diff
@@ -393,8 +393,28 @@ func makeValForInsert(schema *yang.Entry, parent interface{}, keys map[string]st
...
+				rootSch := keySchema
+				for ; rootSch.Parent != nil; rootSch = rootSch.Parent {
+				}
+				pv := util.SplitPath(leafrefPath)
+				v, ok := rootSch.Dir[util.StripModulePrefix(pv[1])]
+				if !ok {
+					return fmt.Errorf("cannot resolve leafref, %s (can't find top-level %s in %v at %s)", leafrefPath, util.StripModulePrefix(pv[1]), rootSch.Dir, rootSch.Name)
+				}
+				if keySchema = v.Find(strings.Join(pv[2:], "/")); keySchema == nil {
+					return fmt.Errorf("cannot find leafref %s from %v", strings.Join(pv[2:], "/"), v.Name)
```
Comment (wenovus) [inline ytypes/list.go:412]: Maybe say "cannot find absolute leafref element" to help with error searching?

File: ytypes/list_test.go [hunk 2]
```diff
@@ -1712,6 +1712,153 @@ func TestSimpleMapKeyValueCreation(t *testing.T) {
...
+						Type: &yang.YangType{Kind: yang.Yleafref, Path: "../baz"},
+						Parent: &yang.Entry{
+							Name: "key",
+						},
+					},
+				},
+			},
+			container:    &simpleStruct{KeyList: map[string]*ListStringStruct{}},
+			errSubstring: `cannot find leafref "../baz"`,
+		},
```
Comment (wenovus) [inline ytypes/list_test.go:1861]: Isn't this a relative leafref test? ../ goes up to the list level right?

File: ytypes/list.go (hunk 1 above)
Comment (robshakir) [reply ytypes/list.go:412]: Sounds good, fixed.

File: ytypes/list_test.go (hunk 2 above)
Comment (robshakir) [reply ytypes/list_test.go:1861]: Yes, sorry, this was a copy-paste error as I was adding coverage for lines that weren't covered. Fixed the testcase name.

**Review by robshakir — COMMENTED** [review]
Thanks for the review.

**Comment by wenovus** [comment]
Thanks, there is a commit to be pushed right?

**Comment by robshakir** [comment]
Yes, apologies! Sat waiting for my SSH key passphrase.

**Review by wenovus — APPROVED** [review]

**Pull request merged**